### DISCUSSION TITLE
How to migrate email templates from database to files? (missing utility)

### DISCUSSION THREAD

**Discussion by jrrdnx** [solspace/craft-freeform#278] [Questions & Answers] [answered]
I have a warning in the Settings -> Diagnostics section to "Update your Email Notification templates. It appears you’re still using the database option for storing Email Notification templates...".

However, in the Settings -> Email Templates section I don't see the utility to actually perform this migration (as referenced in the docs here: https://docs.solspace.com/craft/freeform/v3/setup/settings.html#email-templates).

I currently do NOT have a directory path set in Settings -> Email Templates as I want Freeform to manage all email templates and have no need for overrides, so I'm not sure if this is why I'm seeing that warning. If this is the case, could this warning be hidden since it doesn't apply?

**Comment by kjmartens** [comment, marked as answer]
Hi @jrrdnx,

The migration utility will only display itself once you have set an Email Templates directory path, as it otherwise wouldn't work. The Diagnostics page will continue to warn you about it (without option to dismiss) as this is a change we recommend you take, as Freeform 4 will no longer offer database-based email notification templates.

That said, in case this is holding you back, I want to make sure you're fully aware that File-based email notification templates can be administrated directly inside the control panel the same way database ones can. The experience is seamless, so you wouldn't even be aware that the change happened. 🙂

**Comment by jrrdnx** [comment]
Thanks @kjmartens !

Not holding me back, I'm just trying to clean things up and noticed the warning but was concerned when I didn't see the migration utility.

So currently our emails are managed only through Freeform -> Email Notifications in the Email Body (HTML) field for each notification that's set up. Is this what you're referring to that will be migrated from database to files with v4?

**Reply by kjmartens** [reply]
In Freeform 4, this will not work anymore. We may not be able to automatically migrate them either, since if there is no directory specified, Freeform wouldn't know where to place the file versions of the email notifications. I'm not exactly sure how it'll be handled in Freeform 4, but the smoothest way forward will be to switch/migrate to Files sometime before Freeform 4.

Like I mentioned earlier, once switched over, you won't even be able to tell the difference. Freeform will read the notification template file and break it into individual text inputs, textareas, toggle switches, etc. 🙂

**Reply by jrrdnx** [reply]
For some reason my brain wasn't making the connection until just now. I set the directory path and migrated everything over successfully. Thank you for your help!
